src/compiler/transformers/destructuring.ts

```diff
@@ -331,11 +331,14 @@ namespace ts {
                 location
             );
         }
-        else if (numElements !== 1 && (flattenContext.level < FlattenLevel.ObjectRest || numElements === 0)) {
+        else if (numElements !== 1 && (flattenContext.level < FlattenLevel.ObjectRest || numElements === 0) ||
+                every(elements, (element) => isOmittedExpression(element))) {

```

Comment:
Answer: No. To be spec compliant, we still emit temporaries in js for empty binding patterns in order to attempt to get the elements for the sub-binding patterns/invoke their side-effects (and potentially error if they do not exist).